pages/core-concepts/seed.mdx

```diff
@@ -222,6 +222,132 @@ snaplet.users((x) => [
 ])
 ```
 
+#### Using the generate callback
+
+If you need more control over the value generated for each field, you can also pass a function for that field:
+
+```ts
+snaplet.users([{
+  email: ({ seed }) => copycat.email(seed)
+}])
+```
+
+##### `seed`
+
+A value unique to this particular field, and to this particular model being generated.
+
+`seed` is deterministic in the sense that for each new _session_ (for example, each new run of a seed script using `@snaplet/seed`, or each
+new run of the same [`$transaction`](#client-state)), the generate
+callback will receive the same value for `seed` for the same field of the same model.
+
+```ts
+const run = () =>
+  snaplet.$transaction(async (snaplet) => {
+    await snaplet.customers([
+      {
+        email({ seed }) {
+          return copycat.email(seed);
+        },
+      },
+    ]);
+  });
+
+await run();
+// INSERT INTO public."Customer" ... VALUES (..., 'Autumn_Boyer64366@vague-foal.org', ...);
+
+await run();
+// INSERT INTO public."Customer" ... VALUES (..., 'Autumn_Boyer64366@vague-foal.org', ...);
+```
+
+##### `data`
+
+An object containing the data already generated for a particular model. 
```

Comment:
@jgoux We document the seed API in a sort of narrative style on this page - For documenting these parameters, I break that here a bit and go for something a bit more prose-y or API reference docs style. Is that alright?

Reply:
Yes totally alright